### DISCUSSION TITLE
Error [ERR_PACKAGE_PATH_NOT_EXPORTED] react

### DISCUSSION THREAD

**Discussion by gkielwasser** [nextauthjs/next-auth#2700] [Help] [answered]
### Question 💬

Hi,

I'm using the beta.2 version.

My build give me the following error `Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './react' is not defined by "exports" in my-project\node_modules\next-auth\package.json`. 
The error happen as soon as I encapsulate my app with the SessionProvider provider.

I'm using the following to wrap my app:
```
import { SessionProvider } from "next-auth/react"

<SessionProvider session={pageProps.session}>
      [...]
 </SessionProvider>     
```

Also, I'm not sure what usefull informations I could share about this error.

Thanks for help!

### How to reproduce ☕️

Here is my nextauth dependency: "next-auth": "4.0.0-beta.2"

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
have a reproduction?

**Reply by gkielwasser** [reply]
I'm not sure what I could share as usefull informations. Also not sure how I could set an appropriate reproductible env.
What I see is that the "exports" statement that should export react, is not working in my environnement. I have to say the path and the files exists ("/react"). 
I have node version 14.17.6
Do you have any clues I could check that would be related to that exports ?
Thanks for help

**Reply by balazsorban44** [reply]
4.0.0-beta.2 has the correct exports, so without a reproduction, my only guess is you don't actually have the correct version installed.

try reading the migration docs carefully https://next-auth.js.org/getting-started/upgrade-v4

**Reply by gkielwasser** [reply]
I'm not so sure what is relation but I have updated this morning my next version from v10 to v11. The error is now gone :).

**Reply by balazsorban44** [reply]
No relation at all. next-auth works well with Next 10. The upgrade probably pulled the correct version for next-auth as well in the background.

Glad it worked out though!

**Reply by gkielwasser** [reply]
I can confirm nothing related to a wrong package of next-auth. Tripe check (maybe more) the package version number in node_modules :)
Not keen about those automatic fixes, I have checked my diff on yarn.lock. I have seen lots of @babel/* packages have been updated.
Also the real reason might be the updates of those packages in my environnment during the installation of Next11 version.
[new implementation of the Babel loader for webpack](https://nextjs.org/blog/next-11)
Hope that could help someone

**Reply by Tammerabi** [reply]
Just thought I'd chime in and say I'm having this issue as well.
    "next": "^9.5.3",
    "next-auth": "4.0.0-beta.2",
Tried with both node v13.14.0 and v16.9.0

Edit: seems like upgrading next to 10 fixed it

**Reply by balazsorban44** [reply]
Again, upgrading to next 10 has no effect for this whatsoever. running npm install again probably reinstalled next-auth as well, fixing your problem.

**Reply by gkielwasser** [reply]
I'm not sure what you mean by reinstalled next-auth ?

**Reply by balazsorban44** [reply]
The point is that NextAuth does not depend on any Next 10 specific features, so the reason for the error being gone must be something else than downgrading next

**Comment by gkielwasser** [comment, marked as answer]
So as an answer I would say, double check those dependencies [typescript + beta.2](https://github.com/nextauthjs/next-auth-example/blob/ndom91/update-v4/package.json)

**Reply by sbmsr** [reply]
Thanks @gkielwasser ! this was the exact fix I needed. 

Would be great if we could add something to the docs to mention this according to the version you install. 

If the next-auth team would be willing to guide me on where to add this, I would gladly make this contribution 😄 .

**Reply by balazsorban44** [reply]
You mean this? https://next-auth.js.org/getting-started/upgrade-v4#verify-the-correct-version

**Reply by gkielwasser** [reply]
> Thanks @gkielwasser ! this was the exact fix I needed.
> 
> Would be great if we could add something to the docs to mention this according to the version you install.
> 
> If the next-auth team would be willing to guide me on where to add this, I would gladly make this contribution 😄 .

Hi, 
Could you please specify what specific procedure you applied to fix your issue? 
As @balazsorban44 state, there is No direct link with the Nextjs version (whatever 10 or 11) as next-auth do not rely on any specific feature.

For anyone one jumping on this discussion, **The first** action is for sure to check the next-auth version as state there https://next-auth.js.org/getting-started/upgrade-v4#verify-the-correct-version.

Then, if the package is correct, it means you are still faced to the resolution of the subpackages next-auth export using the "exports" statement in its package.json. 
Even if not directly related, the re-installation of Nextjs seems to align some sub-dependencies (from what I have seen in the comments, the version of Nextjs is not important). In my case I had a diff in my package.lock with @babel packages.

Lastly keep in mind this can be **far** from the real fix as we do not know clearly what is interferring with the proper resolution of the sub-pacakges and lead to this error 'ERR_PACKAGE_PATH_NOT_EXPORTED'

I think it would be nice to identify which packages have changed before/after the applied solution,

Thanks for feedback,

**Reply by saevarb** [reply]
Just to help anyone else passing by: I ran into this error because I hadn't seen the v4 migration docs about there being a bad publish, so I was trying to use beta4. Switching to beta2 fixed this issue.

**Reply by balazsorban44** [reply]
`4.0.0-beta.4` works as well. :wink: If you are having trouble, read this: https://next-auth.js.org/getting-started/upgrade-v4#verify-the-correct-version

**Reply by saevarb** [reply]
> `4.0.0-beta.4` works as well. wink If you are having trouble, read this: https://next-auth.js.org/getting-started/upgrade-v4#verify-the-correct-version

Thanks for the reply! From the docs, it sounds a lot like beta4 isn't valid.

After trying to install beta2 because I was having the package export problem, that problem went away. However, instead, the azure AD provider was broken and I wasted several hours on trying to debug that.. until I tried installing the beta4 again. Doing that un-broke the azure AD provider and let me login again, but no matter what I do, when I deploy the application to Azure App Services with beta4, I get the module export error.

The worst part is that I'm simply not able to reproduce it locally. It only happens when I try to deploy the application. Running the build artifact in a new folder locally doesn't exhibit this issue, and neither does building it and running it isolated in a docker container. 

My conclusion is that there must simply be something funky going on with whatever azure app service is doing. This wouldn't be the first thing, either, since I'm already using a hack like `start:prod": "node_modules/next/dist/bin/next start"` in my package.json file to start it on azure because it simply won't work otherwise.

**Reply by balazsorban44** [reply]
I'm sorry to hear that. Although I'm curious which part of the docs implies that beta.4 wouldn't work? The only broken release is 4.0.0 and installing it will even warn you being deprecated. (I did try to remove it from npm, but I wasn't allowed to)

**Reply by saevarb** [reply]
>  version to be exactly 4.0.0-beta.1 (or beta.2 etc.

I suppose I must have missed the `etc` here - my bad!

Also, I probably should have clarified that my module export error is about the azure-ad provider, not react. 

I also just tried updating the package.json to add the following line to exports:

```
...
"./providers/*": "./providers/*.js",
"./providers/azure-ad": "./providers/azure-ad.js" // <-
...
```


And this does indeed seem to fix the module export error, and I'm finally able to start a sign-in process. It's not the end of my troubles, however, as now I am getting a `CLIENT_FETCH_ERROR`:

```
2021-10-12T12:58:44.008070842Z error: CLIENT_FETCH_ERROR {
2021-10-12T12:58:44.008128343Z   error: FetchError: invalid json response body at https://<website>/<basePath>/api/auth/providers reason: Unexpected token ï»¿ in JSON at position 0
2021-10-12T12:58:44.008136744Z       at /home/site/wwwroot/node_modules/node-fetch/lib/index.js:272:32
2021-10-12T12:58:44.008141244Z       at processTicksAndRejections (internal/process/task_queues.js:97:5)
2021-10-12T12:58:44.008145744Z       at async fetchData (/home/site/wwwroot/node_modules/next-auth/lib/client.js:22:18) {
2021-10-12T12:58:44.008150044Z     type: 'invalid-json'
2021-10-12T12:58:44.008154244Z   },
2021-10-12T12:58:44.008158344Z   path: 'providers'
2021-10-12T12:58:44.008162544Z }
```

The only problem with this error is that if I follow the URL it complains about, it gives me totally valid JSON. 

I have given up on this problem for now and my conclusion regarding azure is the same, and I've talked to my devops people to let me deploy docker containers instead, where I have 100% control of the environment. 

I know this is all mostly irrelevant info but maybe someone else runs into a similar issue.
